### PR TITLE
Historical detection fix

### PR DESCRIPTION
**Pull request by shayzluf**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**
Changing detection flow to detect historical before realtime to be able to remove lookback
Found a bug that caused historical detection not to use cache(underflow)

**Which issues(s) does this PR fix?**

**Other notes for review**

### CODE REVIEW

File: slasher/detection/service.go [hunk 1]
```diff
@@ -88,9 +88,11 @@ func (ds *Service) Start() {
 	if featureconfig.Get().EnableHistoricalDetection {
 		// The detection service runs detection on all historical
 		// chain data since genesis.
-		go ds.detectHistoricalChainData(ds.ctx)
+		ds.detectHistoricalChainData(ds.ctx)
 	}
-
+	// We listen to a stream of blocks and attestations from the beacon node.
+	go ds.beaconClient.ReceiveBlocks(ds.ctx)
```
Comment (0xKiwi) [inline slasher/detection/service.go:94]: Aren't these used in `beaconclient/service.go`?

Comment (shayzluf) [reply slasher/detection/service.go:94]: moved them here to start listening after historical detection ends

**Comment by rauljordan** [comment]
Hi @shayzluf any chance we can write a regression test for this bug?

**Comment by shayzluf** [comment]
Sure @rauljordan

File: slasher/detection/service.go (hunk 1 above)
Comment (0xKiwi) [reply slasher/detection/service.go:94]: Should they be removed then from historical detection?

**Pull request merged**